src/Tickets/Commerce/Order.php

```diff
@@ -371,21 +371,49 @@ public function modify_status( $order_id, $status_slug, array $extra_args = [] )
 			return false;
 		}
 
-		$can_apply = $status->can_apply_to( $order_id, $status );
-		if ( ! $can_apply ) {
-			return $can_apply;
-		}
-
-		$args = array_merge( $extra_args, [ 'status' => $status->get_wp_slug() ] );
+		$this->start_transaction();
```

Comment:
The logic holds here, but the Order class, as I've explained in my other comment, should not have transaction-related thin methods: managing transactions is the `DB` class' responsibility.

A test case can be written to avoid mocking transaction code and actually using transactions (see my other comment).